engine/spverrors/http_response.go

```diff
@@ -10,34 +10,40 @@ import (
 
 // ErrorResponse is searching for error and setting it up in gin context
 func ErrorResponse(c *gin.Context, err error, log *zerolog.Logger) {
-	response, statusCode := getError(err, log)
+	response, statusCode := mapAndLog(err, log)
 	c.JSON(statusCode, response)
 }
 
 // AbortWithErrorResponse is searching for error and abort with error set
 func AbortWithErrorResponse(c *gin.Context, err error, log *zerolog.Logger) {
-	response, statusCode := getError(err, log)
+	response, statusCode := mapAndLog(err, log)
 	c.AbortWithStatusJSON(statusCode, response)
 }
 
-func getError(err error, log *zerolog.Logger) (models.ResponseError, int) {
+func mapAndLog(err error, log *zerolog.Logger) (model models.ResponseError, statusCode int) {
+	model.Code = models.UnknownErrorCode
+	model.Message = "Internal server error"
+	statusCode = 500
+
+	logLevel := log.Warn()
 	var extendedErr models.ExtendedError
 	if errors.As(err, &extendedErr) {
-		return models.ResponseError{
-			Code:    extendedErr.GetCode(),
-			Message: extendedErr.GetMessage(),
-		}, extendedErr.GetStatusCode()
+		model.Code = extendedErr.GetCode()
+		model.Message = extendedErr.GetMessage()
+		statusCode = extendedErr.GetStatusCode()
+		if statusCode >= 500 {
+			logLevel = log.Error()
+		} else {
+			logLevel = log.Info()
+		}
```

Comment:
What's the reason behind it? If we get an error that is different than 5xx wouldn't it be better to read and find in logs if it at least was a Warning level? I found it confusing a little bit at the beginning when I was looking for an error in logs but there was only INFO logs and only then I had to actually look at the response code to figure out there was indeed an error.

Reply:
The reason for assigning a higher log level to `5xx` status codes is that these errors typically `indicate server-side issues`, which are more likely to be caused by problems within "our system". 
- By using a higher log level (like Error) for these, we're emphasizing the need to address potential bugs or misconfigurations on our end.

On the other hand, `4xx` errors generally indicate client-side issues (e.g., incorrect passwords or malformed request data). These errors are part of normal operations, and since they're not related to bugs in our system, we log them at the Info level. 
- Logging them as warnings or errors might mislead developers or users into thinking there's an issue with the system itself, when in fact it's just standard behavior.

Reply:
I agree with @dzolt-4chain although it is a standard behaviour we should consider 4XX as warnings, because it should warn us that someone is misusing our system, and we (or rather maintainers of the production env) should earlier or later address those warnings, either by solving users issue or blocking them.

Reply:
So, is that ok for you?
```
if statusCode >= 500 {
	logLevel = log.Error()
} else {
	logLevel = log.Warn()
}
```

Reply:
Yes exactly